bindings/rust/s2n-tls/src/connection.rs

```diff
@@ -942,3 +942,15 @@ impl Drop for Connection {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    // ensure the config context is send and sync
```

Comment:
The comment here is incorrect. Either that or the test is incorrect :) Presumably this is referring to the Connection Context? But I don't know why that would only be Send and not Sync.

Reply:
The comment is incorrect. And actually I need to remove the `unsafe impl Sync for Connection` as well. The reason it can't be `Sync` is the contained `AsyncCallback` doesn't have `Sync` bounds on it.

Reply:
I guess my next question would then be, _should_ the AsyncCallback have Sync bounds on it? Or is that totally unnecessary. It does sound weird to have something called Async be Sync xD.

Reply:
I don't think so. We should have the same constraints as [tokio::spawn](https://docs.rs/tokio/latest/tokio/task/fn.spawn.html).